src/components/my-widgets-settings-dialog.jsx

```diff
@@ -0,0 +1,728 @@
+import React, { useState, useEffect } from 'react'
```

Comment:
a monster, honkin' component.  this one's going to need some splitting up and optimization.  

1. move any functions we can outside the render loop (looks like a handful here that only operate on their arguments, not relying on scope)
2. `document.anything` is a concern, might need some refs?
3. some functions w/ multiple setStates, try to avoid
4. consider creating components for individual controls